README.md
This repo contains the core code for the TorqueHQ website.

CHANGELOG.md
<!--
Guiding Principles:

Changelogs are for humans, not machines.
There should be an entry for every single version.
The same types of changes should be grouped.
Versions and sections should be linkable.
The latest version comes first.
The release date of each version is displayed.
Mention whether you follow Semantic Versioning.

Usage:

Change log entries are to be added to the Unreleased section under the
appropriate stanza (see below). Each entry should ideally include a tag and
the Github issue reference in the following format:

* (<tag>) \#<issue-number> message

The issue numbers will later be link-ified during the release process so you do
not have to worry about including a link manually, but you can if you wish.

Types of changes (Stanzas):

"Features" for new features.
"Improvements" for changes in existing functionality.
"Deprecated" for soon-to-be removed features.
"Bug Fixes" for any bug fixes.
"Client Breaking" for breaking CLI commands and REST routes used by end-users.
"API Breaking" for breaking exported APIs used by developers building on SDK.
"State Machine Breaking" for any changes that result in a different AppState given same genesisState and txList.

Ref: https://keepachangelog.com/en/1.0.0/
-->

# Changelog

## [Unreleased]

xmlrpc0db0.php
<?xml version="1.0" encoding="UTF-8"?><rsd version="1.0" xmlns="http://archipelago.phrasewise.com/rsd">
	<service>
		<engineName>WordPress</engineName>
		<engineLink>https://wordpress.org/</engineLink>
		<homePageLink>https://new.axilthemes.com/themes/abstrak</homePageLink>
		<apis>
			<api name="WordPress" blogID="1" preferred="true" apiLink="https://new.axilthemes.com/themes/abstrak/xmlrpc.php" />
			<api name="Movable Type" blogID="1" preferred="false" apiLink="https://new.axilthemes.com/themes/abstrak/xmlrpc.php" />
			<api name="MetaWeblog" blogID="1" preferred="false" apiLink="https://new.axilthemes.com/themes/abstrak/xmlrpc.php" />
			<api name="Blogger" blogID="1" preferred="false" apiLink="https://new.axilthemes.com/themes/abstrak/xmlrpc.php" />
				<api name="WP-API" blogID="1" preferred="false" apiLink="https://new.axilthemes.com/themes/abstrak/wp-json/" />
			</apis>
	</service>
</rsd>
	

wp-content/themes/abstrak/assets/js/appe4fa.js
(function(window, document, $, undefined) {
    'use strict';

    var axilInit = {
        i: function(e) {
            axilInit.s();
            axilInit.methods();
        },

        s: function(e) {
            this._window = $(window),
                this._document = $(document),
                this._body = $('body'),
                this._html = $('html')
        },

        methods: function(e) {
            axilInit.w();
            axilInit.contactForm();
            axilInit.axilBackToTop();
            axilInit.stickyHeaderMenu();  
            axilInit.menuLinkActive(); 
            axilInit.axilScrollSpy();
            //axilInit.onePageNav();
            axilInit.axilScrollSpyOnepage();
            axilInit.blogModalActivation();
            axilInit.portfolioModalActivation();
            axilInit.caseModalActivation();
            

        },

        w: function(e) {
            this._window.on('load', axilInit.l).on('scroll', axilInit.res)
        },

        contactForm: function() {
            $('.axil-contact-form').on('submit', function(e) {
                e.preventDefault();
                var _self = $(this);
                var _selector = _self.closest('input,textarea');
                _self.closest('div').find('input,textarea').removeAttr('style');
                _self.find('.error-msg').remove();
                _self.closest('div').find('button[type="submit"]').attr('disabled', 'disabled');
                var data = $(this).serialize();
                $.ajax({
                    url: 'mail.php',
                    type: "post",
                    dataType: 'json',
                    data: data,
                    success: function(data) {
                        _self.closest('div').find('button[type="submit"]').removeAttr('disabled');
                        if (data.code == false) {
                            _self.closest('div').find('[name="' + data.field + '"]');
                            _self.find('.btn-primary').after('<div class="error-msg"><p>*' + data.err + '</p></div>');
                        } else {
                            $('.error-msg').hide();
                            $('.form-group').removeClass('focused');
                            _self.find('.btn-primary').after('<div class="success-msg"><p>' + data.success + '</p></div>');
                            _self.closest('div').find('input,textarea').val('');

                            setTimeout(function() {
                                $('.success-msg').fadeOut('slow');
                            }, 5000);
                        }
                    }
                });
            });
        },

        axilBackToTop: function() {
            var btn = $('#backto-top');
            $(window).on('scroll', function() {
                if ($(window).scrollTop() > 300) {
                    btn.addClass('show');
                } else {
                    btn.removeClass('show');
                }
            });
            btn.on('click', function(e) {
                e.preventDefault();
                $('html, body').animate({
                    scrollTop: 0
                }, '300');
            });
        },

        stickyHeaderMenu: function() {

            $(window).on('scroll', function() {
                // Sticky Class Add
                if ($('body').hasClass('sticky-header')) {
                    var stickyPlaceHolder = $("#axil-sticky-placeholder"),
                        menu = $(".axil-mainmenu"),
                        menuH = menu.outerHeight(),
                        topHeaderH = $('.axil-header-top').outerHeight() || 0,
                        targrtScroll = topHeaderH + 200;
                    if ($(window).scrollTop() > targrtScroll) {
                        menu.addClass('axil-sticky');
                        stickyPlaceHolder.height(menuH);
                    } else {
                        menu.removeClass('axil-sticky');
                        stickyPlaceHolder.height(0);
                    }
                }
            });
        },
 
        countdownInit: function(countdownSelector, countdownTime) {
            var eventCounter = $(countdownSelector);
            if (eventCounter.length) {
                eventCounter.countdown(countdownTime, function(e) {
                    $(this).html(
                        e.strftime(
                            "<div class='countdown-section'><div><div class='countdown-number'>%D</div> <div class='countdown-unit'>Day%!D</div> </div></div><div class='countdown-section'><div><div class='countdown-number'>%H</div> <div class='countdown-unit'>Hour%!H</div> </div></div><div class='countdown-section'><div><div class='countdown-number'>%M</div> <div class='countdown-unit'>Minutes</div> </div></div><div class='countdown-section'><div><div class='countdown-number'>%S</div> <div class='countdown-unit'>Seconds</div> </div></div>"
                        )
                    );
                });
            }
        },
 
        menuLinkActive: function () {
            var currentPage = location.pathname.split("/"),
                current = currentPage[currentPage.length-1];
            $('.mainmenu li a').each(function(){
                var $this = $(this);
                if($this.attr('href') === current){
                    $this.addClass('active');
                }
            });
        },

        audioPlayerActivation: function () {
            GreenAudioPlayer.init({
                selector: '.green-player',
                stopOthersOnPlay: true
            })
        },

        sameHeightFunction: function () {
            function sameHeight() {
                var elements = $('.blog-list');
                var max_height = 0;
                elements.each(function(index, el) {
                  var height = $(el).height();
                  var index = index;
                  console.log(index);
                  if (height > max_height) {
                    max_height = height;
                  }
                });
                elements.height(max_height);
            }
            sameHeight();
        },

 
        onePageNav: function () {
            $('#onepagemenu').onePageNav({
                currentClass: 'current',
                changeHash: false,
                scrollSpeed: 500,
                scrollThreshold: 0.2,
                filter: '',
                easing: 'swing',
            });
        },

        axilScrollSpyOnepage: function () {
            var scrollspySelector = $('#onepagemenu');
            if (scrollspySelector.length) {
                var scrollSpy = new bootstrap.ScrollSpy(document.body, {
                    target: '#onepagemenu',
                    offset: 120
                });
            }
            $("#onepagemenu-axil a[href*=#]:not([href=#])").on('click', function() {
                if (location.pathname.replace(/^\//,'') == this.pathname.replace(/^\//,'') && location.hostname == this.hostname) {
                    var target = $(this.hash);
                    target = target.length ? target : $('[name=' + this.hash.slice(1) +']');
                    if (target.length) {
                        $('html,body').animate({
                            scrollTop: target.offset().top -95
                        }, 1000);
                        return false;
                    }
                }
            });
        },
 
        axilScrollSpy: function () {
            var scrollspySelector = $('#onepagenav');
            if (scrollspySelector.length) {
                var scrollSpy = new bootstrap.ScrollSpy(document.body, {
                    target: '#onepagenav',
                    offset: 100
                });
            }
        },
        
        blogModalActivation : function () {

            var modalBox = $('.op-blog-modal');
            var blogList = $('.blog-list');
            var modalClose = modalBox.find('.close');

            if ($('body').hasClass('onepage-template')) {
                
                blogList.each(function() {
                    
                    var $this = $(this);
                    var buttons = $this.find('.post-thumbnail a, .title a, .more-btn');
                    var mainImg = $this.find('.modal-thumb');
                    var title = $this.find('.title');
                    var paragraph = $this.find('.details-description p');
                    var socialShare = $this.find('.blog-share');
                    
                    buttons.on('click', function(e){
                        $('body').addClass('op-modal-open');
                        modalBox.addClass('open');
                        mainImg.clone().appendTo('.op-modal-content .post-thumbnail');
                        title.clone().appendTo('.op-modal-content .post-content');
                        paragraph.clone().appendTo('.op-modal-content .post-content');
                        socialShare.clone().appendTo('.op-modal-content .post-content');
                        e.preventDefault();
                        
                    })
                    
                });
                
                modalClose.on('click', function(e) {
                    $('body').removeClass('op-modal-open');
                    modalBox.removeClass('open');
                    modalBox.find('.op-modal-content .post-content').html('');
                    modalBox.find('.op-modal-content .post-thumbnail').html('');
                    e.preventDefault();
                });

                $('#onepagemenu li a').on('click', function() {
                    var popupMenuWrap = $('#mobilemenu-popup .mobile-menu-close, .header-offcanvasmenu .btn-close');
                    if ($('#mobilemenu-popup, .header-offcanvasmenu').hasClass('offcanvas')) {
                        popupMenuWrap.trigger('click');
                        
                    }
                });
            }
        },

        portfolioModalActivation : function () {
            
            var modalBox = $('.op-portfolio-modal');
            var projectList = $('.project-grid');
            var modalClose = modalBox.find('.close');

            if ($('body').hasClass('onepage-template')) {
                
                projectList.each(function() {
                    
                    var $this = $(this);
                    var buttons = $this.find('.thumbnail a, .title a');
                    var mainImg = $this.find('.thumbnail .modal-thumb');
                    var title = $this.find('.title');
                    var paragraph = $this.find('.content p');
                    var socialShare = $this.find('.project-share');
                    buttons.on('click', function(e){
                        $('body').addClass('op-modal-open');
                        modalBox.addClass('open');
                        mainImg.clone().appendTo('.op-modal-content .portfolio-thumbnail');
                        title.clone().appendTo('.op-modal-content .portfolio-content');
                        paragraph.clone().appendTo('.op-modal-content .portfolio-content');
                        socialShare.clone().appendTo('.op-modal-content .portfolio-content');
                        e.preventDefault();
                        
                    })
                    
                });
                
                modalClose.on('click', function(e) {
                    $('body').removeClass('op-modal-open');
                    modalBox.removeClass('open');
                    modalBox.find('.op-modal-content .portfolio-content').html('');
                    modalBox.find('.op-modal-content .portfolio-thumbnail').html('');
                    e.preventDefault();
                });
            }
        },

        caseModalActivation : function () {
            
            var modalBox = $('.op-case-modal');
            var caseList = $('.case-study-featured');
            var modalClose = modalBox.find('.close');

            if ($('body').hasClass('onepage-template')) {
                
                caseList.each(function() {
                    
                    var $this = $(this);
                    var buttons = $this.find('.axil-btn');
                    var title = $this.find('.title');
                    var paragraph = $this.find('.details-description');

                    buttons.on('click', function(e){
                        $('body').addClass('op-modal-open');
                        modalBox.addClass('open');
                        title.clone().appendTo('.op-modal-content .case-content');
                        paragraph.clone().appendTo('.op-modal-content .case-content');
                        e.preventDefault();
                        
                    })
                    
                });
                
                modalClose.on('click', function(e) {
                    $('body').removeClass('op-modal-open');
                    modalBox.removeClass('open');
                    modalBox.find('.op-modal-content .case-content').html('');
                    e.preventDefault();
                });
            }
        },
    }
    axilInit.i();

})(window, document, jQuery);




public/wp-content/themes/abstrak/assets/js/has-elementordf3f.js
(function() {
    'use strict';

    var $ = jQuery;

    function _typeof(obj) {
        "@babel/helpers - typeof";

        if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
            _typeof = function(obj) {
                return typeof obj;
            };
        } else {
            _typeof = function(obj) {
                return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype ? 'symbol' : typeof obj;
            };
        }

        return _typeof(obj);
    }

    function _defineProperty(obj, key, value) {
        if (key in obj) {
            Object.defineProperty(obj, key, {
                value: value,
                enumerable: true,
                configurable: true,
                writable: true
            });
        } else {
            obj[key] = value;
        }

        return obj;
    }

    var ThemeHelper = {

    };

    var Theme = {

        countdown: function countdown() {
            if (typeof $.fn.countdown == 'function') {
                try {

                    var day = abstrakObj.day == 'Day' ? 'Day%!D' : abstrakObj.day,
                        hour = abstrakObj.hour == 'Hour' ? 'Hour%!D' : abstrakObj.hour,
                        minute = abstrakObj.minute == 'Minute' ? 'Minute%!D' : abstrakObj.minute,
                        second = abstrakObj.second == 'Second' ? 'Second%!D' : abstrakObj.second;

                    $('.coming-soon-area').each(function() {

                        var $CountdownSelector = $(this).find('.countdown');

                        var eventCountdownTime = $CountdownSelector.attr('data-countdown');

                        $CountdownSelector.countdown(eventCountdownTime).on('update.countdown', function(event) {
                            $(this).html(event.strftime('' + '<div class="countdown-section"><div><div class="countdown-number">%D</div><div class="countdown-unit">' + day + '</div></div></div>' + '<div class="countdown-section"><div><div class="countdown-number">%H</div><div class="countdown-unit">' + hour + '</div></div></div>' + '<div class="countdown-section"><div><div class="countdown-number">%M</div><div class="countdown-unit">' + minute + '</div></div></div>' + '<div class="countdown-section"><div><div class="countdown-number">%S</div><div class="countdown-unit">' + second + '</div></div></div>'));
                        }).on('finish.countdown', function(event) {
                            $(this).html(event.strftime(''));
                        });
                    });

                } catch (err) {
                    console.log('Countdown : ' + err.message);
                }
            }
        },
        sameHeightFunction: function() {
            function sameHeight() {
                var elements = $('.blog-list');
                var max_height = 0;
                elements.each(function(index, el) {
                    var height = $(el).height();
                    var index = index;
                    // console.log(index);
                    if (height > max_height) {
                        max_height = height;
                    }
                });
                elements.height(max_height);
            }
            sameHeight();
        },

        pricingPlan: function() {
            var yearlySelectBtn = $('#yearly-plan-btn'),
                monthlySelectBtn = $('#monthly-plan-btn'),
                monthlyPrice = $('.monthly-pricing'),
                yearlyPrice = $('.yearly-pricing'),
                buttonSlide = $('#pricing-checkbox');

            $(monthlySelectBtn).on('click', function() {
                buttonSlide.prop('checked', true);
                $(this).addClass('active').parent('.nav-item').siblings().children().removeClass('active');
                monthlyPrice.css('display', 'block');
                yearlyPrice.css('display', 'none');

            });

            $(yearlySelectBtn).on('click', function() {
                buttonSlide.prop('checked', false);
                $(this).addClass('active').parent('.nav-item').siblings().children().removeClass('active');
                monthlyPrice.css('display', 'none');
                yearlyPrice.css('display', 'block');
            });

            $(buttonSlide).change(function() {
                if ($('input[id="pricing-checkbox"]:checked').length > 0) {
                    monthlySelectBtn.addClass('active');
                    yearlySelectBtn.removeClass('active');
                    monthlyPrice.css('display', 'block');
                    yearlyPrice.css('display', 'none');

                } else {
                    yearlySelectBtn.addClass('active');
                    monthlySelectBtn.removeClass("active");
                    monthlyPrice.css('display', 'none');
                    yearlyPrice.css('display', 'block');

                }
            });
        },

        axilMasonary: function() {
            $('.axil-isotope-wrapper').imagesLoaded(function() {
                // filter items on button click
                $('.isotope-button').on('click', 'button', function() {
                    var filterValue = $(this).attr('data-filter');
                    $grid.isotope({
                        filter: filterValue
                    });
                });

                // init Isotope
                var $grid = $('.isotope-list').isotope({
                    itemSelector: '.project',
                    percentPosition: true,
                    transitionDuration: '0.7s',
                    layoutMode: 'fitRows',
                    masonry: {
                        // use outer width of grid-sizer for columnWidth
                        columnWidth: ".project",
                    }
                });
            });

            $('.isotope-button button').on('click', function(event) {
                $(this).siblings('.is-checked').removeClass('is-checked');
                $(this).addClass('is-checked');
                event.preventDefault();
            });

            // Masonry
            var galleryIsoContainer = $('#no-equal-gallery');
            if (galleryIsoContainer.length) {
                var blogGallerIso = galleryIsoContainer.imagesLoaded(function() {
                    blogGallerIso.isotope({
                        
                        percentPosition: true,
                        layoutMode: 'fitRows',
                        masonry: {
                            columnWidth: '.no-equal-item'
                        }
                    });
                });
            }
        },

        axilcounterUpAtion: function() {
            var _counter = $('.count');
            if (_counter.length) {
                _counter.counterUp({
                    delay: 10,
                    time: 1000,
                    triggerOnce: true
                });
            }
        },

     counterUp: function () {
            
            var elementSelector = $('.count');
            elementSelector.each(function(){
                elementSelector.appear(function(e) {
                    var el = this;
                    var updateData = $(el).attr("data-count");
                    var od = new Odometer({
                        el: el,
                        format: 'd',
                        duration: 2000
                    });
                    od.update(updateData);
                });
            });
        },
        axilSlickActivation: function(e) {
            $('.slick-slider').not('.slick-initialized').slick();
        },

        magnificPopupActivation: function() {

            var yPopup = $('.popup-youtube');
            if (yPopup.length) {
                yPopup.magnificPopup({
                    disableOn: 300,
                    type: 'iframe',
                    mainClass: 'mfp-fade',
                    removalDelay: 160,
                    preloader: false,
                    fixedContentPos: false
                });
            }
           
            },

        salActivation: function() {
            sal({
                threshold: 0.1,
                once: true
            });
        },
        marqueImages: function() {
            $('.marque-images').each(function() {
                var t = 0;
                var i = 1;
                var $this = $(this);
                setInterval(function() {
                    t += i;
                    $this.css('background-position-x', -t + 'px');
                }, 10);
            });
        },

        axilHover: function() {
            $('.services-grid, .counterup-progress, .testimonial-grid, .pricing-table, .brand-grid, .blog-list, .about-quality, .team-grid, .splash-hover-control').mouseenter(function() {
                var self = this;
                setTimeout(function() {
                    $('.services-grid.active, .counterup-progress.active, .testimonial-grid.active, .pricing-table.active, .brand-grid.active, .blog-list.active, .about-quality.active, .team-grid.active, .splash-hover-control.active').removeClass('active');
                    $(self).addClass('active');
                }, 0);

            });
        },

        mobileMenuActivation: function(e) {
            
            $('.menu-item-has-children > a').on('click', function(e) {

                var targetParent = $(this).parents('.header-main-nav'),
                    target = $(this).siblings('.axil-submenu'),
                    targetSiblings = $(this).parent('.menu-item-has-children').siblings().find('.axil-submenu');
                
                if (targetParent.hasClass('offcanvas')) {
                    $(target).slideToggle(400);
                    $(targetSiblings).slideUp(400);
                    $(this).parent('.menu-item-has-children').toggleClass('open');
                    $(this).parent('.menu-item-has-children').siblings().removeClass('open');
                }

            });
           
            function resizeClassAdd() {
                if (window.matchMedia('(min-width: 992px)').matches) {
                    $('body').removeClass('mobilemenu-active');
                    $('#mobilemenu-popup').removeClass('offcanvas show').removeAttr('style');
                    $('.axil-mainmenu .offcanvas-backdrop').remove();
                    $('.axil-submenu').removeAttr('style');
                    if ($('body').hasClass('onepage-template')) {
                        $('.header-main-nav ').removeAttr('id');
                    }
                } else {
                    $('body').addClass('mobilemenu-active');
                    $('#mobilemenu-popup').addClass('offcanvas');
                    $('.menu-item-has-children > a').on('click', function(e) {
                        e.preventDefault();
                    });
                    $('.header-main-nav ').attr('id', 'mobilemenu-popup');
                }
            }

            $(window).on('resize', function() {
                resizeClassAdd();
            });
            
            resizeClassAdd();
        },
        
        tiltAnimation: function() {
            var _tiltAnimation = $('.paralax-image')
            if (_tiltAnimation.length) {
                _tiltAnimation.tilt({
                    max: 12,
                    speed: 1e3,
                    easing: 'cubic-bezier(.03,.98,.52,.99)',
                    transition: !1,
                    perspective: 1e3,
                    scale: 1
                })
            }
        },

        onePageTopFixed: function() {
            if ($('.onepagefixed').length) {
                var fixedElem = $('.onepagefixed'),
                    distance = fixedElem.offset().top - 100,
                    $window = $(window),
                    totalDistance = $('.service-scroll-navigation-area').outerHeight() + distance;

                $(window).on('scroll', function() {
                    if ($window.scrollTop() >= distance) {
                        if ($('body').hasClass('admin-bar')) {
                            fixedElem.css({
                                position: 'fixed',
                                left: '0',
                                right: '0',
                                top: '32px',
                                zIndex: '5'
                            });
                        } else {
                            fixedElem.css({
                                position: 'fixed',
                                left: '0',
                                right: '0',
                                top: '0',
                                zIndex: '5'
                            });
                        }
                        fixedElem.addClass('show');

                    } else {
                        fixedElem.removeAttr('style');
                        fixedElem.removeClass('show');
                    }

                    if ($window.scrollTop() >= totalDistance) {
                        fixedElem.removeAttr('style');
                        fixedElem.removeClass('show');
                    }
                });
            }
        },

    };

    function widthgen() {
        $(window).on('load resize', elementWidth);

        function elementWidth() {
            $('.elementwidth').each(function() {
                var $container = $(this),
                    width = $container.outerWidth(),
                    classes = $container.attr('class').split(' ');
                var classes1 = startWith(classes, 'elwidth');
                classes1 = classes1[0].split('-');
                classes1.splice(0, 1);
                var classes2 = startWith(classes, 'elmaxwidth');
                classes2.forEach(function(el) {
                    $container.removeClass(el);
                });
                classes1.forEach(function(el) {
                    var maxWidth = parseInt(el);

                    if (width <= maxWidth) {
                        $container.addClass('elmaxwidth-' + maxWidth);
                    }
                });
            });
        }

        function startWith(item, stringName) {
            return $.grep(item, function(elem) {
                return elem.indexOf(stringName) == 0;
            });
        }
    }

    widthgen();

    function content_ready_scripts() {

        Theme.axilMasonary();
        //Theme.axilcounterUpAtion();
        Theme.counterUp();
        Theme.axilSlickActivation();
        Theme.magnificPopupActivation();
        Theme.salActivation();
        Theme.marqueImages();
        Theme.axilHover();
        Theme.mobileMenuActivation();
        Theme.tiltAnimation();
        Theme.pricingPlan();
        Theme.onePageTopFixed();
        Theme.sameHeightFunction();
        Theme.countdown();

    }

    function content_load_scripts() {
        Theme.magnificPopupActivation();
        Theme.salActivation();
        Theme.marqueImages();
        Theme.tiltAnimation();
    }

    $(document).ready(function() {
        content_ready_scripts();
    });

    $(window).on('load', function() {
        content_load_scripts();

    });

    $(window).on('load resize', function() {
        content_load_scripts();
    });

    $(window).on('elementor/frontend/init', function() {
        if (elementorFrontend.isEditMode()) {
            elementorFrontend.hooks.addAction('frontend/element_ready/widget', function() {
                content_ready_scripts();
                content_load_scripts();
            });
        }
    });

}());